src/NServiceBus.Core/Transports/TransportConfig.cs

```diff
@@ -45,6 +45,10 @@ public int MaxRetries
         /// The max throughput for the transport. This allows the user to throttle their endpoint if needed.
         /// </summary>
         [ConfigurationProperty("MaximumMessageThroughputPerSecond", IsRequired = false, DefaultValue = -1)]
+        [ObsoleteEx(
+            TreatAsErrorFromVersion = "6", 
+            RemoveInVersion = "7", 
+            Message = "Message throughput throttling has been removed. Please consult the documentation for further information.")]
```

Comment:
there is a small section [here](http://docs.particular.net/nservicebus/operations/tuning#tuning-throughput) and I'm pretty sure we need to add a statement to the upgrade guide.